### PR TITLE
feat(icon): support for .code-snippet and .code-workspace extensions

### PR DESCRIPTION
**Pull request by remcohaszing**
Use the VSCode icon for `code-snippets` and `.code-workspace` files.

**Changes proposed:**

- [x] Add

### CODE REVIEW

**Comment by remcohaszing** [comment]
I don’t understand the failing test.

```
  1) Specifications of supported extensions: tests
       ensures that
         each supported
           extension
             does NOT have
               same declaration with different 'extensions':
     AssertionError: expected [ Array(1) ] to be empty
      at /home/remco/Projects/vscode-icons/out/test/iconsManifest/specsExtensions.test.js:2:5377
      at Array.forEach (<anonymous>)
      at Context.<anonymous> (out/test/iconsManifest/specsExtensions.test.js:2:5042)
      at process.processImmediate (node:internal/timers:475:21)
```

If I add the extension to the entries that already exist, the icon isn’t applied.

**Comment by voidei** [comment]
@remcohaszing I think I fixed it? I have no idea why exactly what I did works, but I changed it from

```ts
    {
      icon: 'vscode',
      extensions: ['code-workspace'],
      format: FileFormat.svg,
    },
```

to

```ts
    {
      icon: 'vscode',
      extensions: ['code-workspace', 'code-snippets'],
      filenamesGlob: [],
      extensionsGlob: [],
      filename: false,
      format: FileFormat.svg,
    },
```

Tried it with only `code-workspace` in there, only `code-snippets`, and of course both - test runs without issues on my end. As I said I have no clue why that works, but if it works then it works :shrug:

**Comment by remcohaszing** [comment]
That works! Thanks!

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -4861,6 +4887,12 @@ export const extensions: IFileCollection = {
       format: FileFormat.svg,
       disabled: true,
     },
+    {
+      icon: 'vscode-insiders',
+      extensions: ['code-workspace'],
```
Comment (robertohuertasm) [inline src/iconsManifest/supportedExtensions.ts:4892]: Shouldn't we have here also `code-snipperts`?

Comment (remcohaszing) [reply src/iconsManifest/supportedExtensions.ts]: Yes. I’ve updated it to match the other vscode icon configurations.

**Review by robertohuertasm — APPROVED** [review]

**Pull request merged**